### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by danielamoreno699**
### Project LocalStorage

_In this milestone **project of LocalStorage**, we:_

-Added a function in the main.js file that stores the form data in the local storage.
-Added a function that pre fills the inputs with values stored in the local storage when the page reloads

### CODE REVIEW

**Review by didierganthier — COMMENTED** [review]
Hi Team,

While you put in a lot of hard work in this project, unfortunately, I cannot proceed with the review of your code.



## Invalid Code Review Request 🚫

- You have submitted a project that doesn't have the JavaScript linters. For this project, you should add the [JavaScript + HTML + CSS linter](https://github.com/microverseinc/linters-config/tree/master/html-css-js).

Your Code Review Request will be marked as invalid in your Dashboard, but don't worry you can always try again! 💪

Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

--------

_Invalid Code Review Request **does not count** towards the code reviews limit._

File: main.js [hunk 1]
```diff
@@ -47,61 +44,41 @@ const closeMenu = () => {
...
+  const formData = JSON.parse(dataString);
+  const formControl = document.getElementById('form');
+
+  for (let i = 0; i < formControl.elements.length; i + 1) {
+    const element = formControl.elements[i];
+    const value = formData[element.name];
+    if (value !== undefined) {
+      element.value = value;
+    }
+  }
```
Comment (BertrandConxy) [inline main.js:80]: - [ ] Good job so far 👏  From the project requirements, the form data are supposed to be saved in local storage as the user typing or changes input content NOT when the user submits the form which is how your site is behaving. Please use the form input event `Change` or OnChange in other to implement this feature appropriately. This event is fired every time there is change in input value.
That means even if the user reloads page without submitting the form, the data that they entered should be auto-filled 😃

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @danielamoreno699 and @federicaulzurrun  , 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Linters are passing ✔️ 
- Data is preserved and persistent 👍 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**